### ISSUE TITLE
mlpack Embedding Layer and Categorical Cross Entropy Loss

### ISSUE THREAD

**Issue by ShaonReza** [mlpack/mlpack#3143]
I am new to C++ and MLPack. I want to convert the below keras-python implementation to MLPack C++ implementation. But seems like - Embedding Layer as well as Categorical Cross Entropy Loss are missing here - https://www.mlpack.org/doc/mlpack-3.4.2/doxygen/index.html. Any suggestion in this regard will be appreciated.

1. self.model = Sequential()
2. self.model.add(Embedding(vocab_size, seq_len, input_length=seq_len + 1)
3. self.model.add(LSTM(50, return_sequences=True))
4. self.model.add(LSTM(50))
5. self.model.add(Dense(50, activation='relu'))
6. self.model.add(Dense(vocab_size, activation='softmax'))
7. self.model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])

**Comment by hello-fri-end** [comment]
Hi @ShaonReza! `Embedding layer` in mlpack is an [alias](https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/lookup.hpp#L142) for the [Lookup layer](https://mlpack.org/doc/mlpack-3.1.0/doxygen/classmlpack_1_1ann_1_1Lookup.html) and, as far as I know, `Categorical Cross Entropy Loss` is just softmax followed by [Cross Entropy Loss ](https://mlpack.org/doc/mlpack-3.1.0/doxygen/classmlpack_1_1ann_1_1CrossEntropyError.html).
